src/org/jetbrains/java/decompiler/modules/decompiler/sforms/SSAProcessor.java

```diff
@@ -0,0 +1,964 @@
+package org.jetbrains.java.decompiler.modules.decompiler.sforms;
+
+import org.jetbrains.java.decompiler.code.CodeConstants;
+import org.jetbrains.java.decompiler.modules.decompiler.exps.*;
+import org.jetbrains.java.decompiler.modules.decompiler.stats.*;
+import org.jetbrains.java.decompiler.modules.decompiler.vars.VarVersionEdge;
+import org.jetbrains.java.decompiler.modules.decompiler.vars.VarVersionNode;
+import org.jetbrains.java.decompiler.modules.decompiler.vars.VarVersionPair;
+import org.jetbrains.java.decompiler.modules.decompiler.vars.VarVersionsGraph;
+import org.jetbrains.java.decompiler.struct.StructMethod;
+import org.jetbrains.java.decompiler.struct.gen.MethodDescriptor;
+import org.jetbrains.java.decompiler.util.*;
+
+import java.util.*;
+
+import static org.jetbrains.java.decompiler.modules.decompiler.sforms.VarMapHolder.mergeMaps;
+
+class SSAProcessor {
```

Comment:
I think this should be renamed "SFormsConstructor" and made abstract. The reasoning for the name change is that the "Processor" suffix has a specific connotation in the existing QF code and old FF code, so this would add confusion as it's not making structural changes to the statement like the other processors.

Reply:
I actually wanted to make another subclass for "virtual" ssa(u) processing that collects information but doesn't actually change the var exprents so that'd be useful there

Reply:
useful where?

Reply:
as in, being able to subclass and such- i personally don't think you should be able to construct a raw SFormsConstructor